partshop-backend/src/main/java/hu/kosztaauto/partshop/service/AlkatreszService.java

```diff
@@ -1,19 +1,58 @@
 package hu.kosztaauto.partshop.service;
 
-import hu.kosztaauto.partshop.database.Alkatresz;
+import hu.kosztaauto.partshop.dto.AlkatreszDTO;
+import hu.kosztaauto.partshop.dto.AlkatreszWithWarehouseDTO;
+import hu.kosztaauto.partshop.dto.RaktarDTO;
+import hu.kosztaauto.partshop.model.Alkatresz;
 import hu.kosztaauto.partshop.repository.AlkatreszRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 public class AlkatreszService {
 
     @Autowired
     private AlkatreszRepository alkatreszRepository;
 
-    public List<Alkatresz> getAllParts() {
-        return alkatreszRepository.findAll();
+    public List<AlkatreszDTO> getAllItems() {
+        return alkatreszRepository.findAll().stream()
+                .map(this::convertToAlkatreszDTO)
+                .collect(Collectors.toList());
     }
-}
+
+    public List<AlkatreszWithWarehouseDTO> getAllWarehousesByItem() {
+        return alkatreszRepository.findAll().stream()
+                .map(this::convertToAlkatreszWithWarehouseDTO)
+                .collect(Collectors.toList());
+    }
+
+    private AlkatreszDTO convertToAlkatreszDTO(Alkatresz alkatresz) {
```

Comment:
Isnt there any automapping package for java?

Reply:
There is but this was my solution, it's fine

Reply:
Maybe I'll update it in a future version